tests/python_functional/functional_tests/applications/test_applications_with_app_parser.py

```diff
@@ -0,0 +1,64 @@
+    ("syslog-raw", "<190>30: foo: *Apr 29 13:58:46.411: %SYS-6-LOGGINGHOST_STARTSTOP: Logging to host 192.168.1.239 stopped - CLI initiated", "${.app.name}", "cisco"),
+    ("syslog-raw", "<190>31: foo: *Apr 29 13:58:46.411: %SYS-6-LOGGINGHOST_STARTSTOP: Logging to host 192.168.1.239 started - CLI initiated<189>32: 0.0.0.0: *Apr 29 13:59:12.491: %SYS-5-CONFIG_I: Configured from console by console<189>33: 0.0.0.0: *Apr 29 13:59:26.415: %SYS-5-CONFIG_I: Configured from console by console<189>34: 0.0.0.0: *Apr 29 13:59:56.603: %SYS-5-CONFIG_I: Configured from console by console^[[<189>35: *Apr 29 14:00:16.059: %SYS-5-CONFIG_I: Configured from console by console", "${.app.name}", "cisco"),
+    ("syslog-raw", "<190>32: foo: *Apr 29 13:58:46.411: %SYSMGR-STANDBY-3-SHUTDOWN_START: The System Manager has started the shutdown procedure.", "${.app.name}", "cisco"),
+    ("syslog-raw", "<134>{\\\"count\\\": 1, \\\"supporting_data\\\": {\\\"data_values\\\": [\\\"x.x.x.x\\\", \\\"user@domain.com\\\"], \\\"data_type\\\": \\\"user\\\"}, \\\"organization_unit\\\": \\\"domain/domain/Domain Users/Enterprise Users\\\", \\\"severity_level\\\": 2, \\\"category\\\": null, \\\"timestamp\\\": 1547421943, \\\"_insertion_epoch_timestamp\\\": 1547421943, \\\"ccl\\\": \\\"unknown\\\", \\\"user\\\": \\\"user@domain.com\\\", \\\"audit_log_event\\\": \\\"Login Successful\\\", \\\"ur_normalized\\\": \\\"user@domain.com\\\", \\\"_id\\\": \\\"936289\\\", \\\"type\\\": \\\"admin_audit_logs\\\", \\\"appcategory\\\": null}", "${.app.name}", "netskope"),
+    ("syslog-raw", "<159>Dec 19 10:48:57 EST 10.203.28.21 vendor=Websense product=Security product_version=7.7.0 action=permitted severity=1 category=153 user=- src_host=10.64.134.74 src_port=62189 dst_host=mail.google.com dst_ip=74.125.224.53 dst_port=443 bytes_out=197 bytes_in=76 http_response=200 http_method=CONNECT http_content_type=- http_user_agent=Mozilla/5.0_(Windows;_U;_Windows_NT_6.1;_enUS;_rv:1.9.2.23)_Gecko/20110920_Firefox/3.6.23 http_proxy_status_code=200 reason=- disposition=1034 policy=- role=8 duration=0 url=https://mail.google.com", "${.app.name}", "websense")
+]
+
+@pytest.mark.parametrize("topic, input_message, template, expected_value", test_parameters_raw,
+                         ids=list(map(str, range(len(test_parameters_raw)))))
+def test_application_raw(config, syslog_ng, topic, input_message, template, expected_value):
+    config.add_include("scl.conf")
+
+    generator_source = config.create_example_msg_generator(num=1, template="\"" + input_message + "\\n\"")
+    app_parser = config.create_app_parser(topic=topic)
+    file_destination = config.create_file_destination(file_name="output.log", template="\"" + template + "\\n" + "\"")
+    config.create_logpath(statements=[generator_source, app_parser, file_destination])
+
+    syslog_ng.start(config)
+    assert file_destination.read_log() == expected_value + "\n"
+
```

Comment:
If possible I would eliminate the "\n"-s from the testcases.

Reply:
I pushed a patch to the top which moves the `\n`-s to the parameters, instead of adding them in the test. Please check if this is better. Now the parameters are a little bit trickier, because the new lines need to be escaped differently in the template variable and the expected variable: for example
```
([cisco message example], "${.app.name}\\n", "cisco\n")
```
Of course the same symptom existed in the previous version (the root cause is the same), just expressed in the test instead of the test parameters.